keanu-examples/bayesianMethodsForHackers/src/main/java/examples/ChallengerDisaster.java

```diff
@@ -0,0 +1,104 @@
+package examples;
+
+import com.google.common.primitives.Booleans;
+import data.ChallengerData;
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.bool.probabilistic.Flip;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Does temperature correlate with defects?
+ *
+ * http://nbviewer.jupyter.org/github/CamDavidsonPilon/Probabilistic-Programming-and-Bayesian-Methods-for-Hackers/blob/master/Chapter2_MorePyMC/Ch2_MorePyMC_PyMC3.ipynb#Example:-Challenger-Space-Shuttle-Disaster-
+ */
+public class ChallengerDisaster {
+    public static ChallengerPosteriors run() {
+        GaussianVertex beta = new GaussianVertex(0, 0.001);
+        GaussianVertex alpha = new GaussianVertex(0, 0.001);
+
+        double[] temps = new double[ChallengerData.LAUNCH_EVENTS.length];
+        IntStream.range(0, temps.length).forEach(i -> temps[i] = ChallengerData.LAUNCH_EVENTS[i][0]);
+        boolean[] defects = new boolean[ChallengerData.LAUNCH_EVENTS.length];
+        IntStream.range(0, defects.length).forEach(i -> defects[i] = ChallengerData.LAUNCH_EVENTS[i][1] == 1);
+
+        DoubleVertex temp = new ConstantDoubleVertex(temps);
+        DoubleVertex logisticOutput = createLogisticFunction(beta, alpha, temp);
+        Flip defect = new Flip(logisticOutput);
+        defect.observe(defects);
+
+        BayesianNetwork net = new BayesianNetwork(defect.getConnectedGraph());
+
+        NetworkSamples networkSamples = MetropolisHastings.getPosteriorSamples(net, net.getLatentVertices(), 120000)
+                .drop(100000).downSample(2);
+
+        ChallengerPosteriors cp = new ChallengerPosteriors();
+        cp.networkSamples = networkSamples;
+        cp.vertices = net.getLatentVertices();
+        cp.alphaSamples = networkSamples.get(alpha).asList().stream().map(s -> s.scalar()).collect(Collectors.toList());
```

Comment:
might be worth having a "toScalar" function which converts a list of samples to scalars.

This could either be a util function in this project, or ideally a feature request to Keanu, we can add a function which returns the samples as a list of scalars

Reply:
you can create a feature request on the GraphOS engineering jira, or talk to Caleb